template/src/features/counter/Counter.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import {
+  decrement,
+  increment,
+  incrementByAmount,
+  selectCount,
+} from './counterSlice';
+import styles from './Counter.module.css';
+
+export function Counter() {
+  const count = useSelector(selectCount);
+  const dispatch = useDispatch();
+  const [incrementAmount, setIncrementAmount] = useState('2');
```

Comment:
If I keep this a number, I get typescript errors when I use `setIncrementAmount` in the `input` below

Reply:
Right, it needs to be converted from a string to a number somewhere.  Either keep it as a string in the component and convert to a number right before you dispatch, or convert it to a number in the change handler and store it as a number.

Reply:
I think it should be kept as a number in state, since that's what the data type represents logically.

Reply:
If I keep it as a number in the state, it kind of messes with what the user can type into the textbox. For example, you can't backspace because it converts an empty string into "0". 

Currently it gets converted to a number right before the dispatch.

Reply:
The non-TypeScript templates uses a number in state though. Does it also have this issue?

Reply:
I think we should change the JS template to be consistent. JS doesn't have an issue currently at run-time because it changes it to a string once the user starts typing.

Reply:
Right, I want both templates consistent in terms of actual code as much as possible.

Reply:
I wonder if it can be a `number?` to allow for the empty case

Reply:
Or keep as string and use it to demonstrate simple validation

Reply:
I think the simplest thing to do is just keep it as a string.  Even `number?` will have problems, because it will restrict what the user can type, which could be confusing. e.g. I can't type `1.` if I want to increment by a fractional number.